util/types/time.go

```diff
@@ -1047,7 +1041,11 @@ func ParseTime(str string, tp byte, fsp int) (Time, error) {
 		return Time{Time: ZeroTime, Type: tp}, errors.Trace(err)
 	}
 
-	return t.Convert(tp)
+	t.Type = tp
+	if err = t.check(); err != nil {
```

Comment:
Convert's semantic is not best fit for here.
Convert means I already have something, wants convert it to another, 
but here I just want to new something, and check it's valid.

@hanfei1991